### PR TITLE
[Scala 3] Support extension methods in hover

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, hover would show extension methods as a simple method. Now, we try to show the extension signature next to the actual extension method signature. This isn't yet perfect as it will not work if a method has a using param as the first one. This is not a big issue though as parameter in both places will work exactly the same.

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 1]
```diff
@@ -117,31 +144,100 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
+  ): (List[List[Symbol]], List[List[Symbol]]) = {
+    def isUsingClause(params: List[Symbol]): Boolean = params match {
+      case sym :: _ => sym.is(Flags.Given)
+      case _ => false
+    }
+
+    def isTypeParamClause(params: List[Symbol]): Boolean = params match {
+      case sym :: _ => sym.is(Flags.TypeParam)
+      case _ => false
+    }
```
Comment (dos65) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:203]:
```suggestion
    def headHasFlag(params: List[Symbol], flag: Flag): Boolean = params match {
      case sym :: _ => sym.is(flag)
      case _ => false
    }
       
    def isUsingClause(params: List[Symbol]): Boolean  =
       headHasFlag(params, Flags.Given)

    def isTypeParamClause(params: List[Symbol]): Boolean =
       headHasFlag(params, Flags.TypeParam)
```

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 2]
```diff
@@ -117,31 +144,100 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
+        if (gsym.name.isRightAssocOperatorName) {
+          val (leadingTyParamss, rest1) = paramss.span(isTypeParamClause)
+          val (leadingUsing, rest2) = rest1.span(isUsingClause)
+          val (rightTyParamss, rest3) = rest2.span(isTypeParamClause)
+          val (rightParamss, rest4) = rest3.splitAt(1)
+          val (leftParamss, rest5) = rest4.splitAt(1)
+          val (trailingUsing, rest6) = rest5.span(isUsingClause)
+          if (leftParamss.nonEmpty)
+            leadingTyParamss ::: leadingUsing ::: leftParamss ::: trailingUsing ::: rightTyParamss ::: rightParamss ::: rest6
+          else paramss // it wasn't a binary operator, after all.
```
Comment (dos65) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:221]: Is there a test case on right accoc extension method?

**Review by dos65 — COMMENTED** [review]
Cool! It was annoying for me seeing `extension (...) def ..` in completions. Thx for fixing that!

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 3]
```diff
@@ -62,13 +62,12 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
-  def hoverDetails(
+  def hoverDetailString(
       sym: Symbol,
       history: ShortenedNames,
-      info: Type,
-      addFullDef: Boolean = true
+      info: Type
   )(using Context): String = {
-    val fullDef = if (addFullDef) fullDefinition(sym, info) else ""
+    val fullDef = fullDefinition(sym, info)
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:70]: Yeah, I was thinking about it, but did not want to touch too much in this PR. I guess I already did that :sweat_smile: 

Overall, I think it's useful to separate these for completions and hovers, I always had issues figuring it out how to make both work.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:221]: Not currently, I will add that.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 4]
```diff
@@ -117,31 +144,100 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
     }.iterator
+    val paramLabelss = label(methodParams)
+    val extLabelss = label(extParams)
+
+    val returnType = typeString(shortType(gtpe.finalResultType, shortenedNames))
+    val extensionSignature = paramssString(extLabelss, extParams)
+
+    val extensionSignatureString =
+      if (extParams.nonEmpty) extensionSignature.mkString("extension ", "", " ")
+      else ""
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:175]: Actually, not sure if I understood that.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Added!

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 1 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Thanks, fixed.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Ok, I merged the two methods and reworked it all a bit. Let me know what you think!

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala [hunk 5]
```diff
@@ -23,71 +24,92 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
...
+      typeSymbol.owner.fullName.stripModuleClassSuffix.toString
+
+    def name: String = nameString(sym)
+
+    def parentPackagesString: String = {
+      val owners = for {
+        sym <- sym.ownersIterator
+        if !(sym.isRoot || sym.isEmptyPackage)
+      } yield super.nameString(sym.name)
+      owners.toList.reverse.mkString(".")
```
Comment (dos65) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:56]: wdty about moving it into MtagsEnrichments?
There is a similar method: `fullNameBackticked` so the common part of `take to the root or empry` might be reused

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (dos65) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala:170]: This is pretty minor - `extensionSignatureString` is used only if `onlyMethodParams = false`.
Now it's at line 177

**Review by dos65 — APPROVED** [review]
LGTM!
Other comments are minor - feel free to ignore

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (dos65) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Cool! IMO, now it's easier to understand rendering rules)

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 5 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Actually, we can just use `fullNameBackticked` :sweat_smile: Removed the method altogether.

File: mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala]: Changed it into a def

**Pull request merged**